tree/ntuple/v7/src/RField.cxx

```diff
@@ -1313,56 +1318,24 @@ ROOT::Experimental::RCollectionClassField::RCollectionClassField(std::string_vie
    fProperties = fProxy->GetProperties();
    if (fProxy->HasPointers())
       throw RException(R__FAIL("collection proxies whose value type is a pointer are not supported"));
-   if (fProperties & TVirtualCollectionProxy::kIsAssociative)
-      throw RException(R__FAIL("associative collections not supported"));
 
-   fIFuncsRead = RCollectionIterableOnce::GetIteratorFuncs(fProxy.get(), true /* readOnly */);
-   fIFuncsWrite = RCollectionIterableOnce::GetIteratorFuncs(fProxy.get(), false /* readOnly */);
-
-   std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> itemField;
-   if (auto valueClass = fProxy->GetValueClass()) {
-      // Element type is a class
-      itemField = RFieldBase::Create("_0", valueClass->GetName()).Unwrap();
-   } else {
-      switch (fProxy->GetType()) {
-      case EDataType::kChar_t:   itemField = std::make_unique<RField<char>>("_0"); break;
-      case EDataType::kUChar_t:  itemField = std::make_unique<RField<std::uint8_t>>("_0"); break;
-      case EDataType::kShort_t:  itemField = std::make_unique<RField<std::int16_t>>("_0"); break;
-      case EDataType::kUShort_t: itemField = std::make_unique<RField<std::uint16_t>>("_0"); break;
-      case EDataType::kInt_t:    itemField = std::make_unique<RField<std::int32_t>>("_0"); break;
-      case EDataType::kUInt_t:   itemField = std::make_unique<RField<std::uint32_t>>("_0"); break;
-      case EDataType::kLong_t:
-      case EDataType::kLong64_t:
-         itemField = std::make_unique<RField<std::int64_t>>("_0");
-         break;
-      case EDataType::kULong_t:
-      case EDataType::kULong64_t:
-         itemField = std::make_unique<RField<std::uint64_t>>("_0");
-         break;
-      case EDataType::kFloat_t:  itemField = std::make_unique<RField<float>>("_0"); break;
-      case EDataType::kDouble_t: itemField = std::make_unique<RField<double>>("_0"); break;
-      case EDataType::kBool_t:   itemField = std::make_unique<RField<bool>>("_0"); break;
-      default:
-         throw RException(R__FAIL("unsupported value type"));
-      }
-   }
-   fItemSize = itemField->GetValueSize();
```

Comment:
The `fItemSize` field is not declared in the `RCollectionClassField` base class but (only) used in the derived classes, I think. It would be better than to move it to the derived classes.

Reply:
With the latest updates, `fItemSize` is initialized in the constructors of the derived classes (`RCollectionClassField` and `RSetField`) but used in `AppendImpl` and `ReadGlobalImpl` of the `RProxiedCollectionField` base class. This is not ideal, but I'm not sure yet what an alternative option could be, considering that for e.g. map fields we might need to want to store e.g. `fKeySize` and `fValueSize` instead and override the read and write methods. What do you think?

Reply:
I agree that the current state is not ideal.

One possible solution would be to have a `RProxiedCollectionField` constructor (perhaps with `protected` visibility) that takes an already constructed `itemField` whose size we can query (as in [line 1387](https://github.com/root-project/root/pull/12948/files#diff-1a2cfcc415e3af5e10c390121775f2e2865ae357dacee02772a4872cd7620e52L1387)) and attach it.  Common parts (e.g. checks for unsupported cases, initializing `fIFuncs(Read|Write)`...) would stay in that constructor.

Then, the constructor that doesn't take an `itemField` should be able to construct one by querying the collection proxy ([lines 1360 - 1386](https://github.com/root-project/root/pull/12948/files#diff-1a2cfcc415e3af5e10c390121775f2e2865ae357dacee02772a4872cd7620e52L1360-L1386)) and delegate the rest to the other constructor.